helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -591,8 +591,11 @@ public void addCurrentStateChangeListener(org.apache.helix.CurrentStateChangeLis
   @Override
   public void addCustomizedStateRootChangeListener(CustomizedStateRootChangeListener listener,
       String instanceName) throws Exception {
-    addListener(listener, new Builder(_clusterName).customizedStatesRoot(instanceName),
-        ChangeType.CUSTOMIZED_STATE_ROOT, new EventType[]{EventType.NodeChildrenChanged});
+    PropertyKey propertyKey = _dataAccessor.keyBuilder().customizedStatesRoot(instanceName);
+    if (_zkclient.exists(propertyKey.getPath())) {
+      addListener(listener, new Builder(_clusterName).customizedStatesRoot(instanceName),
+          ChangeType.CUSTOMIZED_STATE_ROOT, new EventType[]{EventType.NodeChildrenChanged});
+    }
```

Comment:
Do you think it would be useful to log (warn) if this evaluates to false?

Reply:
Did we say that if participant's version is older than the controller version, it's fine? If so then I think we may not need to log it.